tokio/src/task/mod.rs

```diff
@@ -69,11 +74,18 @@ pub(crate) trait Schedule: Send + Sync + Sized + 'static {
     fn schedule(&self, task: Task<Self>);
 }
 
+/// Marker trait indicating that a scheduler can only schedule tasks which
+/// implement `Send`.
+///
+/// Schedulers that implement this trait may not schedule `!Send` futures. If
+/// trait is implemented, the corresponding `Task` type will implement `Send`.
+pub(crate) trait ScheduleSend: Schedule + Send + Sync {}
```

Comment:
The name here is weird to me. `ScheduleSend` reads like something that _can_ schedule something that is `Send`, not something that _cannot_ schedule something that is `!Send`.

Reply:
@jonhoo WDYT of `RequiresSend`? That was @carllerche's original suggestion. I can change it back if people think that's clearer.

Reply:
It's mind-bending to read the trait bounds now, since `ScheduleSend` is something that _restricts_ implementations, rather than something that has additional features (the way traits usually work). I'm not sure how to fix that.

Reply:
`RequiresSend` is still pretty weird. I don't quite know how to put my finger on it. I think it's more the _fact_ that it is inherently a negative bound than its particular name. May be hard to fix at all. What I really want is for the bounds to all say `where S: !ScheduleUnsend`.